bundles/org.eclipse.swt/Eclipse SWT Tests/win32/org/eclipse/swt/widgets/WidgetWin32Tests.java

```diff
@@ -0,0 +1,25 @@
+package org.eclipse.swt.widgets;
+
+import static org.junit.Assert.assertEquals;
+
+import org.eclipse.swt.*;
+import org.eclipse.swt.internal.*;
+import org.junit.*;
+
+public class WidgetWin32Tests extends Win32AutoscaleTestBase {
+
+	@Test
+	public void testWidgetZoomShouldChangeOnZoomLevelChange() {
+		int zoom = DPIUtil.getDeviceZoom();
+		int scaledZoom = zoom * 2;
+
+		Button button = new Button(shell, SWT.PUSH);
+		button.setBounds(0, 0, 200, 50);
+		button.setText("Widget Test");
+		button.setBackground(shell.getDisplay().getSystemColor(SWT.COLOR_CYAN));
+		shell.open();
+		changeDPIZoom(scaledZoom);
+		assertEquals("The Zoom Level should be updated for button on zoom change event on its shell", scaledZoom, button.getZoom());
```

Comment:
Nit: I would explicitly check for this to make the test clearer and more future-proof:

```suggestion
		assertEquals("The initial zoom is wrong", zoom, button.getZoom()); //pre-condition
		changeDPIZoom(scaledZoom);
		assertEquals("The Zoom Level should be updated for button on zoom change event on its shell", scaledZoom, button.getZoom());
```